Content.Shared/Mind/SharedMindSystem.cs

```diff
@@ -294,6 +294,22 @@ public bool TryRemoveObjective(EntityUid mindId, MindComponent mind, int index)
         return true;
     }
 
+    public bool TryGetObjectiveComp<T>(EntityUid mindId, MindComponent? mind, [NotNullWhen(true)] out T? objective)
+    {
+        if (Resolve(mindId, ref mind))
```

Comment:
killer if (!Resolve(mindId, ref mind)) return

also the signature should be `EntityUid mindId, [NotNullWhen(true)] out T? objective, MindComponent? mind = null`

Reply:
noob you have to update every usage of it too

Reply:
i don't `!Resolve` because I would have to do `objective = default` twice

Reply:
just do `objective = null;` before resolve